includes/class-wc-stripe-intent-controller.php

```diff
@@ -856,4 +858,84 @@ public function is_mandate_data_required( $selected_payment_type ) {
 
 		return $is_stripe_link_enabled || $is_sepa_debit_payment;
 	}
+
+	/**
+	 * Creates and confirm a setup intent with the given payment method ID.
+	 *
+	 * @param string $payment_method The payment method ID (pm_).
+	 *
+	 * @throws WC_Stripe_Exception If the create intent call returns with an error.
+	 *
+	 * @return array
+	 */
+	public function create_and_confirm_setup_intent( $payment_method ) {
+		// Determine the customer managing the payment methods, create one if we don't have one already.
+		$user        = wp_get_current_user();
+		$customer    = new WC_Stripe_Customer( $user->ID );
+		$customer_id = $customer->update_or_create_customer();
+
+		$setup_intent = WC_Stripe_API::request(
+			[
+				'customer'       => $customer_id,
+				'confirm'        => 'true',
+				'payment_method' => $payment_method,
+			],
+			'setup_intents'
+		);
+
+		if ( ! empty( $setup_intent->error ) ) {
+			throw new WC_Stripe_Exception( $setup_intent->error->message );
```

Comment:
Should this be 
```suggestion
			throw new WC_Stripe_Exception( print_r( $setup_intent->error, true ), $setup_intent->error->message );
```

It seems like you're intending the message received from Stripe to be the localized message.

That's at least my understanding and how it looks like it's used in other places.

Reply:
I just tested this and have confirmed this is an issue. I faked the setup intent request and passed it an invalid customer id and it appeared like this: 

<img width="1122" alt="Screenshot 2023-12-18 at 3 41 16 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/a77f5f90-9b17-4942-8654-358ca1ad96e7">

With the suggested changes above it looks like this: 

<img width="1167" alt="Screenshot 2023-12-18 at 3 42 58 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/ca933b4a-d449-4dca-9b11-5377aa215581">

Reply:
another way I've just replicated this is to use the card `4000000000000101` (CVC check fails)

Reply:
Ah, thanks @james-allan ! I confirmed the issue and confirmed your suggested changes are correct! Thanks!!